hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMTransactionInfo.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership. The ASF
+ * licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * <p>http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * <p>Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.hadoop.hdds.scm.ha;
+
+import com.google.common.base.Preconditions;
+import org.apache.hadoop.hdds.StringUtils;
+
+import java.util.Objects;
+
+import static org.apache.hadoop.ozone.OzoneConsts.TRANSACTION_INFO_SPLIT_KEY;
+
+final public class SCMTransactionInfo {
```

Comment:
To reuse `OMTransactionInfo`, my current answer is probably no.

Though both `OMTransactionInfo` and `SCMTransactionInfo` seems are doing the same thing for now, decoupling is good when there is a need that either `OMTransactionInfo` and `SCMTransactionInfo` need to add new/drop existing fields while the other does not need. Especially in this time that SCM snapshot is still under development. 

I can create a JIRA though to track the merge of two transaction info.

Reply:
Sorry, I'm not clear why we can not reuse OMTransactionInfo. They has the same fields.

Reply:
I think we are better to reuse `OMTransactionInfo` after merge HDDS-2823 back to master.

Not reusing `OMTransactionInfo` is to try to reduce the risk that have conflicts after syncing master int HDDS-2823. Also, to reuse `OMTransactionInfo`, I think we'd better to update the name to `TransactionInfo`, which will be a source of complicates, unless we can change it in the master branch first.

To conclude a bit, we can reuse it after merging SCM HA back to master, the goal is try to reduce potential conflicts if there could be any.

SCM HA is in developing branch anyway and I am sure we will need to clean up lots of stuff eventually. As long as we have JIRAs to not lose track of such work.

Reply:
`OMTransactionInfo` is written by Bharat, since SCM and OM has a quite similar requirement, we may have a talk with him, to determine whether OM and SCM can share the same `TransactionInfo` and the `Codec`. @amaliujia  What do you think ?

Reply:
Sure we can do that. I will need add some changes in main branch, and then sync that to HDDS-2823 and then reuse in SCM.

I think we should do that separately as that will take time. I have created https://issues.apache.org/jira/browse/HDDS-4660 to track this effort.